fqdn/__init__.py

```diff
@@ -26,15 +26,23 @@ class FQDN:
     length of a label is 63 bytes without the leading length byte.
     """
 
-    FQDN_REGEX = re.compile(
+    STRICT_FQDN_REGEX = re.compile(
         r"^((?!-)[-A-Z\d]{1,63}(?<!-)\.)+(?!-)(?=.*[A-Z])([-A-Z\d]{1,63})?(?<!-)\.?$",
         re.IGNORECASE,
     )
+    LOOSE_FQDN_REGEX = re.compile(
+        r"^((?![-_])[-_A-Z\d]{1,63}(?<!-)\.)*((?!-)[-A-Z\d]{1,63}(?<!-)\.)(?!-)(?=.*[A-Z])([-A-Z\d]{1,63})?(?<!-)\.?$",
+        re.IGNORECASE,
+    )
 
-    def __init__(self, fqdn):
+    def __init__(self, fqdn, strict=True):
```

Comment:
This is easily clobberable by accident. I'm a big fan of [PEP3102's keyword-only args](https://www.python.org/dev/peps/pep-3102/#specification) for cases like this. 

Could we switch this to `def __init__(self, fqdn, *, strict=True):` ?  :)

Reply:
100% down, never know what people like I always do that